Jellyfin.Plugin.ChapterSegments/Configuration/PluginConfiguration.cs

```diff
@@ -8,7 +10,43 @@ namespace Jellyfin.Plugin.ChapterSegments.Configuration;
 public class PluginConfiguration : BasePluginConfiguration
 {
     /// <summary>
-    /// Gets or sets the regular expression pattern for detecting the intro segment type.
+    /// Gets or sets the user provided regex text for Intros.
     /// </summary>
-    public string? IntroPattern { get; set; } = "intro|opening";
-}
+    public string? CustomMappingIntro { get; set; } = "intro|opening|^OP$";
+
+    /// <summary>
+    /// Gets or sets the user provided regex text for Commercials.
+    /// </summary>
+    public string? CustomMappingPreview { get; set; } = "preview|next time on|next on|sneek peak";
+
+    /// <summary>
+    /// Gets or sets the user provided regex text for Previews.
+    /// </summary>
+    public string? CustomMappingRecap { get; set; } = "recap|last time on|last on|previously on";
+
+    /// <summary>
+    /// Gets or sets the user provided regex text for Recaps.
+    /// </summary>
+    public string? CustomMappingOutro { get; set; } = "outro|closing|ending|^ED$";
+
+    /// <summary>
+    /// Gets or sets the user provided regex text for Outros.
+    /// </summary>
+    public string? CustomMappingCommercial { get; set; } = "break|ad|advertisement|intermission";
+
+    /// <summary>
+    /// Gets the regular expressions with a mapping of their respective Segment types.
+    /// </summary>
+    public IReadOnlyList<(MediaSegmentType Type, string Regex)> Patterns()
+    {
+        List<(MediaSegmentType Type, string Regex)> list = new List<(MediaSegmentType Type, string Regex)>()
+        {
+            { MediaSegmentType.Intro, CustomMappingIntro },
+            { MediaSegmentType.Commercial, CustomMappingCommercial },
+            { MediaSegmentType.Preview, CustomMappingPreview },
+            { MediaSegmentType.Recap, CustomMappingRecap },
+            { MediaSegmentType.Outro, CustomMappingOutro },
+        };
+        return list;
+    }
```

Comment:
```suggestion
    public IReadOnlyList<(MediaSegmentType Type, string? Regex)> Patterns()
    {
        return
        [
            (MediaSegmentType.Intro, CustomMappingIntro),
            (MediaSegmentType.Commercial, CustomMappingCommercial),
            (MediaSegmentType.Preview, CustomMappingPreview),
            (MediaSegmentType.Recap, CustomMappingRecap),
            (MediaSegmentType.Outro, CustomMappingOutro),
        ];
    }
```

Since the `Regex` patterns are nullable we'll need to update the calling code to handle potential null values